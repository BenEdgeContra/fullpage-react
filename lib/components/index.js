import Fullpage from './Fullpage';
import Slide from './Slide';
import HorizontalSlider from './HorizontalSlider';

export {
  Fullpage,
  Slide,
  HorizontalSlider
};
