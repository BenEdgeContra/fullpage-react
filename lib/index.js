import { Fullpage, HorizontalSlider, Slide } from './components';

export {
  Fullpage,
  Slide,
  HorizontalSlider
};
