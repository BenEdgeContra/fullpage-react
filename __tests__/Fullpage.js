import React from 'react';
import { Fullpage, Slide, HorizontalSlider } from '../lib';
import renderer from 'react-test-renderer';

const fullPageOptions = {
	scrollSensitivity: 7,
	touchSensitivity: -3,
	scrollSpeed: 500,
	resetSlides: true,
	hideScrollBars: true,
	enableArrowKeys: true,
	breakpoint: 375
};

function createNodeMock(element) {
  return {
    addEventListener: () => {}
  };
}

describe('FullpageReact', () => {

  describe('with Slides', () => {
    beforeEach(() => {
      const verticalSlides = [
        <Slide style={{backgroundColor: 'blue'}}><p>1</p></Slide>,
        <Slide style={{backgroundColor: 'pink'}}><p>2</p></Slide>
      ];
      fullPageOptions.slides = verticalSlides;
    });

    it('should support vertical slides', () => {
      const component = renderer.create(
        <Fullpage {...fullPageOptions}/>,
        {createNodeMock}
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('with HorizontalSlides', () => {
    beforeEach(() => {
      const horizontalSliderProps = {};
      const horizontalSlides = [
        <Slide style={{backgroundColor: 'red'}}><p>Horizontal 1</p></Slide>,
        <Slide style={{backgroundColor: 'yellow'}}><p>Horizontal 2</p></Slide>,
        <Slide style={{backgroundColor: 'green'}}><p>Horizontal 3</p></Slide>
      ];
      horizontalSliderProps.slides = horizontalSlides;

      const verticalSlides = [
        <Slide style={{backgroundColor: 'blue'}}><p>1</p></Slide>,
        <HorizontalSlider {...horizontalSliderProps} infinite={true} name={'horizontal1'}/>
      ];
      fullPageOptions.slides = verticalSlides;
    });

    it('should support horizontal slides', () => {
      const component = renderer.create(
        <Fullpage {...fullPageOptions}/>,
        {createNodeMock}
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

});
