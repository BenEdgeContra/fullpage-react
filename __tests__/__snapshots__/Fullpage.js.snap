// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullpageReact with HorizontalSlides should support horizontal slides 1`] = `
<div
  className="Fullpage"
>
  <div
    className="VerticalSlide"
    data-horizontal-slider={null}
    data-slide="VerticalSlide"
    id={undefined}
    style={
      Object {
        "backgroundColor": "blue",
        "height": 768,
        "overflow": "hidden",
        "position": "relative",
        "width": 1024,
      }
    }
  >
    <p>
      1
    </p>
  </div>
  <div
    className="HorizontalSlider"
    data-name="horizontal1"
    data-slide="HorizontalSlider"
    id={undefined}
    style={
      Object {
        "height": 768,
        "margin": "0",
        "overflowX": "hidden",
        "padding": "0px",
        "position": "relative",
        "whiteSpace": "nowrap",
        "width": 1024,
      }
    }
  >
    <div
      className="HorizontalSlide"
      data-horizontal-slider="horizontal1"
      data-slide="HorizontalSlide"
      id={undefined}
      style={
        Object {
          "backgroundColor": "red",
          "height": 768,
          "overflow": "hidden",
          "position": "relative",
          "width": 1024,
        }
      }
    >
      <p>
        Horizontal 1
      </p>
    </div>
    <div
      className="HorizontalSlide"
      data-horizontal-slider="horizontal1"
      data-slide="HorizontalSlide"
      id={undefined}
      style={
        Object {
          "backgroundColor": "yellow",
          "height": 768,
          "overflow": "hidden",
          "position": "relative",
          "width": 1024,
        }
      }
    >
      <p>
        Horizontal 2
      </p>
    </div>
    <div
      className="HorizontalSlide"
      data-horizontal-slider="horizontal1"
      data-slide="HorizontalSlide"
      id={undefined}
      style={
        Object {
          "backgroundColor": "green",
          "height": 768,
          "overflow": "hidden",
          "position": "relative",
          "width": 1024,
        }
      }
    >
      <p>
        Horizontal 3
      </p>
    </div>
  </div>
</div>
`;

exports[`FullpageReact with Slides should support vertical slides 1`] = `
<div
  className="Fullpage"
>
  <div
    className="VerticalSlide"
    data-horizontal-slider={null}
    data-slide="VerticalSlide"
    id={undefined}
    style={
      Object {
        "backgroundColor": "blue",
        "height": 768,
        "overflow": "hidden",
        "position": "relative",
        "width": 1024,
      }
    }
  >
    <p>
      1
    </p>
  </div>
  <div
    className="VerticalSlide"
    data-horizontal-slider={null}
    data-slide="VerticalSlide"
    id={undefined}
    style={
      Object {
        "backgroundColor": "pink",
        "height": 768,
        "overflow": "hidden",
        "position": "relative",
        "width": 1024,
      }
    }
  >
    <p>
      2
    </p>
  </div>
</div>
`;
